tests/test_components.py

```diff
@@ -538,114 +543,128 @@ def test_parse(self):
+        version = PageVersionFactory(content__page=page)  # newer version
+        component = self._get_collections_obj(pages={"page1": page})
 
-    def test_add_version(self):
-        """
-        Test that adding a version to a collection creates 
-        a moderation request which references that version
-        """
-        collection_data = {
-            "author": self.user,
-            "workflow": self.wf1,
-            "name": "Collection 1",
-        }
-        collection = self.component.get_or_create(collection_data)
-        version = self.component.add_version(collection, self.version)
-        moderation_request = collection.moderation_requests.first()
-        stored_version = moderation_request.version
-        # check the stored version
-        self.assertEqual(self.version, stored_version)
-        # check the stored content
-        self.assertEqual(moderation_request.version.content, self.version.content)
-
-    def test_integration(self):
-        component = self.component
-        component.raw_data = {
-            "collection1": {
-                "pages": ["page1"],
-                "name": "Collection 1",
-                "user": "user1",
-                "workflow": "wf1",
-            }
-        }
         component.parse()
-        self.assertEqual(ModerationCollection.objects.count(), 1)
-        collection = component.data["collection1"]
-        self.assertEqual(collection.name, "Collection 1")
+
+        request = ModerationRequest.objects.get()  # newly created request
+        self.assertEqual(request.collection.name, "Collection 1")
+        # TODO: This should probably be equal to the latest version but
+        # is not. Temporarily testing it is equal to the old version until
+        # we confirm this should be fixed
+        from djangocms_versioning.models import Version
+        old_version = Version.objects.get(state=ARCHIVED)
+        self.assertEqual(request.version, old_version)
+        # self.assertEqual(request.version, version)
+        self.assertEqual(request.author, self.user)
+        # It appears the language field is not currently used by
+        # moderation, hence probably why this is left empty
+        self.assertEqual(request.language, "")
+        self.assertTrue(request.is_active)
+        # TODO: Should this default to the date of the import?
+        self.assertEqual(request.date_sent, now())
+
+    def test_parse_does_not_add_moderation_request_if_collection_already_exists(self):
+        """If a ModerationCollection with that name exists, do not add any
+        moderation requests to it."""
+        existing_collection = ModerationCollectionFactory(name="Collection 1")
+        page = PageContentWithVersionFactory().page
+        component = self._get_collections_obj(pages={"page1": page})
+
+        component.parse()
+
+        # TODO: Discussed with Riz that this should probably be the expected
```

Comment:
Think that's a yes

Reply:
Yes it should be latest version.
Sure, it can default to the date of the import
The next todo says "Need to verify with Krzysztof".